java/vector/src/main/java/org/apache/arrow/vector/TimeMilliVector.java

```diff
@@ -125,9 +127,9 @@ public LocalDateTime getObject(int index) {
     if (isSet(index) == 0) {
       return null;
     }
-    org.joda.time.LocalDateTime ldt = new org.joda.time.LocalDateTime(get(index),
-            org.joda.time.DateTimeZone.UTC);
-    return ldt;
+    final int millis = valueBuffer.getInt(index * TYPE_WIDTH);
+    // TODO: this doesn't seem right, time not from epoch
+    return DateUtility.getLocalDateTimeFromEpochMilli(millis);
```

Comment:
This change here is a change in API. It will break all applications using this getObject API.